tests/bsim/run_parallel: Fail if an explicitly listed test is missing & fix missing tests

aescolar opened the pull request:
Instead of just printing a warning, fail (return != 0 to the shell) if a test is missing, or a pattern does not match any test

tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso.sh was split in two in 872b19321f11e272bdfb3fa07811d6482b0868b9 but for the 53 and 5340 it was forgotten in the list of tests to run.
Let's fix it.

aescolar committed:
tests/bsim/run_parallel: Fail if an explicitly listed test is missing

Instead of just printing a warning, fail (return != 0 to the shell)
if a test is missing, or a pattern does not match any test.

Also, warn (and return != 0) if no tests are found when SEARCH_PATH
is provided.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/bsim/bluetooth: Fix BIS test for 54L15 an 5340

tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso.sh
was split in two in 872b19321f11e272bdfb3fa07811d6482b0868b9
but for the 54L15 and 5340 it was forgotten in the list of
tests to run.
Let's fix it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

cvinayak reviewed (changes requested):
Update commit message? `but for the 53 and 5340 it was forgotten in the list of
tests to run.`  should be `but for the 54L15 and 5340 it was forgotten in the list of
tests to run.`

Also, fix the bsim job failure?

aescolar committed:
tests/bsim: UART: Do not attempt to run empty set of multidevice tests

We never got to add any multidevice tests to the UART driver tests.
So let's not try to run that empty set in CI.

(we have some multidevice UART tests for BT).

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

cvinayak commented on tests/bsim/run_parallel.sh:
```diff
@@ -51,7 +52,8 @@ elif [ -n "${TESTS_LIST}" ]; then
 else
 	SEARCH_PATH="${SEARCH_PATH:-.}"
-	all_cases=`find ${SEARCH_PATH} -name "*.sh" | grep -Ev "${sh_filter}"`
+	all_cases=`find ${SEARCH_PATH} -name "*.sh" | grep -Ev "${sh_filter}"` \
+	          | (echo "No tests found")
```
Not sure what is right?! (am not a bash script expert)

```suggestion
	all_cases=`find ${SEARCH_PATH} -name "*.sh" | grep -Ev "${sh_filter}" | echo "No tests found"`
```

cvinayak reviewed (approved):
Not a bash expert, but approving the inclusions of the tests.

Thalley commented:
> Not a bash expert, but approving the inclusions of the tests.

Github's Copilot can provide additional explanation: 
![image](https://github.com/user-attachments/assets/b9f8069a-8f27-4e48-b6f3-888de7230556)
 :D
